android-client/Espoo/app/src/main/res/layout/activity_login.xml

```diff
@@ -0,0 +1,137 @@
+        android:layout_marginTop="8dp"
+        android:text="@string/register_now"
+        android:textStyle="bold"
+        app:layout_constraintStart_toEndOf="@+id/textView2"
+        app:layout_constraintTop_toBottomOf="@+id/textView" />
+
+    <TextView
+        android:id="@+id/textView4"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginLeft="16dp"
+        android:layout_marginTop="32dp"
+        android:text="@string/login_username"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView2" />
+
+    <EditText
+        android:id="@+id/editTextTextPersonName2"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:layout_marginStart="16dp"
+        android:layout_marginLeft="16dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginRight="16dp"
+        android:ems="10"
+        android:inputType="textPersonName"
+        android:minHeight="48dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView4" />
+
+    <TextView
+        android:id="@+id/textView5"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginLeft="16dp"
+        android:layout_marginTop="24dp"
+        android:text="@string/login_password"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/editTextTextPersonName2" />
+
+    <EditText
+        android:id="@+id/editTextTextPassword"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginLeft="16dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginRight="16dp"
+        android:ems="10"
+        android:inputType="textPassword"
+        android:minHeight="48dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView5" />
+
+    <Button
+        android:id="@+id/button"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="16dp"
+        android:layout_marginLeft="16dp"
+        android:layout_marginTop="24dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginRight="16dp"
+        android:text="@string/login_btn"
+        app:backgroundTint="@color/teal_700"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/editTextTextPassword"
+        tools:ignore="TextContrastCheck" />
```

Comment:
You go reuse many times this button style. You can to do in global style?

Reply:
Nice question! But I think the `<Button/>` tag already do that. Basically, the most of attributes is related to button location in screen. The other attributes like text and color could be encapsulated in a new component of course but I don't think it would be very useful. 
In Android development, unlike web development, we build screens without look the screen xml code, because we use the Android Studio User Interface to do that. There is no difference between the xml code made manually and the code generated by UI. The difference is the speed. Build Android screens through the UI is 10x faster. In my early years as Android developer I was only building screens writing code because I was learning, but today, I use juste the UI.
(I hope you'll read @iurynogueira, it took me half an hour to write)

Reply:
Nice text, personal improvement suggestions

Nice question! But I think the <Button/> tag already do that. Basically, ~~the~~ most of attributes **are** related to button location in screen. ~~The~~ other attributes like text and color could be encapsulated in a new component of course but I don't think it would be very useful.
In Android development, unlike web development, we build screens without look the screen xml code, because we use ~~the~~ Android Studio User Interface to do that. There is no difference between ~~the~~ xml code made manually and ~~the~~ code generated by UI. The difference is ~~the~~ speed. Build Android screens through ~~the~~ UI is 10x faster. In my early years as Android developer I was only building screens writing code because I was learning, but today, I use juste the UI.
(I hope you'll read @iurynogueira, it took me half an hour to write)

Reply:
Thank you for the feedback @edimossilva . If all my friend did that, may Mark Zuckerberg would no longer be the CEO of Facebook.

Reply:
I readed everything @amiapmorais, i think that create a component is a good idea to reuse in application and keep a patter without preocupation.

Reply:
@amiapmorais I have no opinion about this, but if you decided to do it, I suggest creating an issue and do it in other PR and resolve this conversation so far

Reply:
when there is the need to have a component for a button I will do it